requests/models.py

```diff
@@ -414,7 +414,7 @@ def prepare_body(self, data, files, json=None):
         content_type = None
         length = None
 
-        if json is not None:
+        if not data and json is not None:
```

Comment:
Sure, will change.

On Wednesday 09 September 2015 04:57 AM, Ian Cordasco wrote:
>
> In requests/models.py 
> <https://github.com/kennethreitz/requests/pull/2763#discussion_r38992512>:
>
> > @@ -414,7 +414,7 @@ def prepare_body(self, data, files, json=None):
> >          content_type = None
> >          length = None
> >
> > -        if json is not None:
> > +        if not data and json is not None:
>
> Since we apparently can't use |if data is None| can we compare to an 
> empty dictionary? Explicit is so much better than implicit.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/kennethreitz/requests/pull/2763/files#r38992512>.
>